index.js

```diff
@@ -132,6 +132,12 @@ RedirectableRequest.prototype._processResponse = function (response) {
 		var redirectUrl = url.resolve(this._currentUrl, location);
 		debug('redirecting to', redirectUrl);
 		Object.assign(this._options, url.parse(redirectUrl));
+		// Fix: 'this._options.headers.Host' may be inconsistent with the new value of 'this._options.host', which may
```

Comment:
Yes, that sounds like a good idea. I modified the pull request in a way that it clears the Host property.